python/packages/isce3/focus/cheby_windows.py

```diff
@@ -0,0 +1,70 @@
+from enum import Enum, unique
+import numpy as np
+from isce3.core import KernelF32, ChebyKernelF32
+
+
+# helper classes for fitting
+
+class CosineWindowF32(KernelF32):
+    def __init__(self, pedestal_height: float = 1.0):
+        width = 1.0
+        KernelF32.__init__(self, width)
+        self.pedestal_height = pedestal_height
+        self.a0 = (1.0 + pedestal_height) / 2
+        self.a1 = 1.0 - self.a0
+        self.freq = 2 * np.pi
+
+    def __call__(self, t: float) -> np.float32:
+        if not (-0.5 <= t <= 0.5):
+            return np.float32(0)
+        return np.float32(self.a0 - self.a1 * np.cos(self.freq * (t + 0.5)))
+
+
+class KaiserWindowF32(KernelF32):
+    def __init__(self, beta: float = 0.0):
+        width = 1.0
+        KernelF32.__init__(self, width)
+        self.beta = beta
+        self.scale = 1.0 / np.i0(beta)
+
+    def __call__(self, t: float) -> np.float32:
+        if not (-0.5 <= t <= 0.5):
+            return np.float32(0)
+        x = self.beta * np.sqrt(1 - 4 * t * t)
+        return np.float32(np.i0(x) * self.scale)
+
+
+@unique
+class WindowKind(str, Enum):
+    KAISER = "kaiser"
+    COSINE = "cosine"
```

Comment:
NBD, but I like to add something like this to `str`/`Enum` classes so that they behave a little bit more like Python 3.11 [`StrEnum`s](https://docs.python.org/3/library/enum.html#enum.StrEnum).

```suggestion
    COSINE = "cosine"
    
    def __str__(self) -> str:
        return self.value
```

Without overriding the `__str__` method, you'll get something like this when converting the enum to a string:

```python
>>> class WindowKind(str, Enum):
...     KAISER = "kaiser"
...     COSINE = "cosine"
...
>>> str(WindowKind.KAISER)
'WindowKind.KAISER'
```

With this change, you'll instead get this:

```python
>>> class WindowKind(str, Enum):
...     KAISER = "kaiser"
...     COSINE = "cosine"
...
...     def __str__(self):
...         return self.value
...
>>> str(WindowKind.KAISER)
'kaiser'
```